contoso_manufacturing/developer/webapp-decode/yolov8.py

```diff
@@ -156,22 +185,41 @@ def draw_detections(self, img, box, score, class_id):
         cv2.putText(img, label, (label_x, label_y), cv2.FONT_HERSHEY_SIMPLEX, 1.2, (0, 0, 0), 1, cv2.LINE_AA)
 
     def run(self):
-        if(self.verbose):
-            print("Running detection...")
+        while not self.stopped:
+            if(self.verbose):
+                print("Running detection...")
+
+            self.frame_number += 1
+            # If mod = 0, i will get the frame and skip it
+            if self.frame_number % self.skip_rate == 0:
+                if not self.frame_queue.empty():
+                    self.cap = self.frame_queue.get()
+                continue
+            
+            if not self.frame_queue.empty():
```

Comment:
If the queue is empty, we should probably log that so we have visibility to what is going on..

Should we have this thread pause briefly (5-10ms maybe) to allow the frame grabber to catch up (otherwise the main thread might show artificially high CPU usage that could be misleading in some scenarios)?